### PR TITLE
#182631487-USERS SHOULD BE ABLE TO SEE ARTICLE READING STATS

### PR DESCRIPTION
**Pull request by ChrisOnsando**
#### What does this Pull Request do?
- [ ] Users should be able to see article reading stats
#### Description of Task to be completed?
- [ ] Users should be able to see article reading stats
#### How should this be manually tested?
- [ ] Via Postman
- [ ] GET:api/articles/statistics/
#### Any background context you want to provide?
- [ ] For an article statistics report is important to the author to enable the author of the article see the impact of the article to the readers
#### What are the relevant Pivotal Tracker Story?
- [ ] (#182631487)-https://www.pivotaltracker.com/story/show/182631487
#### Screenshots (if appropriate)
Screenshot from 2022-08-12 13-19-53
#### Questions:
- [ ] N/A

### CODE REVIEW

File: app/articles/serializers.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ class ArticleSerializer(serializers.ModelSerializer):
     )
     favourite_count = serializers.SerializerMethodField()
     unfavourite_count = serializers.SerializerMethodField()
+    views = serializers.IntegerField(read_only=True)
```
Comment (mainamuriuki274) [inline app/articles/serializers.py:61]: How are the views being updated?

File: app/articles/tests/test_views.py [hunk 2]
```diff
@@ -82,6 +87,14 @@ def test_create_article(self, upload_resource: Any) -> None:
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
         self.assertEqual(Article.objects.count(), count + 1)
 
+    def test_get_all_articles(self) -> None:
+        response = self.client.get(
+            reverse("article-list"),
+            data=self.data,
+            **self.bearer_token,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (mainamuriuki274) [inline app/articles/tests/test_views.py:96]: assert more than once

File: app/articles/views.py [hunk 3]
```diff
@@ -118,6 +119,13 @@ class ArticleUnFavouriteView(generics.UpdateAPIView):
     renderer_classes = (JSONRenderer,)
 
 
+class ArticleStatsView(generics.ListCreateAPIView):
```
Comment (mainamuriuki274) [inline app/articles/views.py:122]: why are you using ListCreate instead of ListAPIView?

Comment (mainamuriuki274) [inline app/articles/views.py:122]: What if I want statistics for a particular article not all?

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: app/articles/views.py (hunk 3 above)
Comment (ChrisOnsando) [reply app/articles/views.py:122]: Initially I had a queryset

Comment (ChrisOnsando) [reply app/articles/views.py:122]: Making changes in the files to implement this

**Pull request closed without merging**